docs/demonstration-auto-tag-new-orders-with-reconciliation/script.liquid

```diff
@@ -0,0 +1,111 @@
+{% assign order_processed_tag = "processed-by-autotagger" %}
+
+{% assign order_ids_to_process = array %}
+
+{% if event.topic contains "shopify/orders/" %}
+  {% if event.preview %}
+    {% assign order = hash %}
+    {% assign order["admin_graphql_api_id"] = "gid://shopify/Order/1234567890" %}
+  {% endif %}
+
+  {% assign order_tags = order.tags | split: ", " %}
+  {% if order_tags contains order_processed_tag %}
+    {% log message: "Order is already processed; skipping" %}
+  {% else %}
+    {% assign order_ids_to_process[order_ids_to_process.size] = order.admin_graphql_api_id %}
+  {% endif %}
+{% elsif event.topic contains "mechanic/scheduler/" or event.topic == "mechanic/user/trigger" %}
+  {% assign cursor = nil %}
+  
+  {% assign orders_query_parts = array %}
+  {% assign orders_query_parts[0] = order_processed_tag | json | prepend: "-tag:" %}
+  {% assign orders_query_parts[1] = "now - 15 minutes" | date: "%Y-%m-%dT%H:%M:%SZ", tz: "UTC" | json | prepend: "created_at:>=" %}
+  {% assign orders_query = orders_query_parts | join: " " %}
+  
+  {% log orders_query: orders_query %}
+  
+  {% for n in (0..100) %}
+    {% capture query %}
+      query {
+        orders(
+          first: 250
+          after: {{ cursor | json }}
+          query: {{ orders_query | json }}
+        ) {
+          pageInfo {
+            hasNextPage
+          }
+          edges {
+            cursor
+            node {
+              id
+              tags
+            }
+          }
+        }
+      }
+    {% endcapture %}
+  
+    {% assign result = query | shopify %}
+  
+    {% if event.preview %}
+      {% capture result_json %}
+        {
+          "data": {
+            "orders": {
+              "edges": [
+                {
+                  "node": {
+                    "id": "gid://shopify/Order/1234567890",
+                    "tags": []
+                  }
+                }
+              ]
+            }
+          }
+        }
+      {% endcapture %}
+  
+      {% assign result = result_json | parse_json %}
+    {% endif %}
+  
+    {% for order_edge in result.data.orders.edges %}
+      {% comment %}
```

Comment:
Is that possible in this context? Could this query return a record that should have been exclude by tag, while at the same time listing out that the order does in fact have the tag in question?

Reply:
Yeah, it came up in my testing. Shopify's API for this stuff is backed by a search index of some kind, and (at least on the dev store I used here) there was some lag between the moment the order was tagged, and the moment when the search results started responding to that tag. Less than 10 seconds, but it was still there.